contrib/adsk/libraries/adsklib/adsklib_defs.mtlx

```diff
@@ -144,15 +144,15 @@
     <input name="surface_cutout" type="color3" value="1, 1, 1" />
     <input name="surface_albedo" type="color3" value="1, 1, 1" />
     <input name="opaque_albedo" type="color3" value="1, 1, 1" />
-    <input name="surface_roughness" type="float" value="0" />
-    <input name="surface_anisotropy" type="float" value="0" />
-    <input name="surface_rotation" type="float" value="0" />
+    <input name="surface_roughness" type="float" value="0" uimin="0.0" uimax="1.0" />
+    <input name="surface_anisotropy" type="float" value="0" uimin="0.0" uimax="1.0" />
+    <input name="surface_rotation" type="float" value="0" uimin="0.0" uimax="180.0" />
     <input name="opaque_f0" type="float" value="1" />
     <input name="opaque_translucency" type="boolean" value="false" />
     <input name="opaque_mfp_modifier" type="color3" value="1, 1, 1" />
-    <input name="opaque_mfp" type="float" value="1" unittype="distance" />
+    <input name="opaque_mfp" type="float" value="1" unittype="distance" uimin="0.0" uisoftmax="100.0" />
```

Comment:
perhaps add a comment or actual default input unit for the mean free path. I believe we use mm

Reply:
I'm unclear how units work in Mtlx. Let's assume we put the unit as mm here. When an app wants to use inches will it explicitly convert before calling the graph? Becasue inside the graph there is no conversion.
Or does MaterialX magically converts things in their APIs?
My assumptions were that angles are in degees if not otherwise specified, while other distances are in scene units if omitted. But if here we expect mm only then I guess putting mm here is a good thing.